drivers/firmware/tisci/tisci.c

```diff
@@ -0,0 +1,1599 @@
+		return -EINVAL;
+	}
+
+	if (k_sem_take(&response_ready_sem, K_MSEC(dev_data->desc.max_rx_timeout_ms)) != 0) {
+		LOG_ERR("Timeout waiting for response");
+		return -ETIMEDOUT;
+	}
+
+	if (xfer->rx_message.size > dev_data->desc.max_msg_size) {
+		LOG_ERR("rx_message.size [ %d ] > max_msg_size\n", xfer->rx_message.size);
+		return -EINVAL;
+	}
+
+	if (rx_message.size < xfer->rx_message.size) {
+		LOG_ERR("rx_message.size [ %d ] < xfer->rx_message.size\n", rx_message.size);
+		return -EINVAL;
+	}
+
+	memcpy(xfer->rx_message.buf, rx_message.buf, xfer->rx_message.size);
+	hdr = (struct tisci_msg_hdr *)xfer->rx_message.buf;
+
+	/* Sanity check for message response */
+	if (hdr->seq != dev_data->seq) {
+		LOG_ERR("HDR seq != data seq [%d != %d]\n", hdr->seq, dev_data->seq);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int tisci_do_xfer(const struct device *dev, struct tisci_xfer *xfer)
+{
+	if (!dev) {
+		return -EINVAL;
+	}
+
+	const struct tisci_config *config = dev->config;
+	struct mbox_msg *msg = &xfer->tx_message;
+	int ret;
+
+	ret = mbox_send_dt(&config->mbox_tx, msg);
+	if (ret < 0) {
+		LOG_ERR("Could not send (%d)\n", ret);
+		return ret;
+	}
+
+	/* Get response if requested */
+	if (xfer->rx_message.size) {
+		ret = tisci_get_response(dev, xfer);
+		if (ret) {
+			return ret;
+		}
+		if (!tisci_is_response_ack(xfer->rx_message.buf)) {
+			LOG_ERR("TISCI Response in NACK\n");
+			return -ENODEV;
+		}
+	}
+
+	return 0;
+}
+
+static int tisci_init(const struct device *dev)
+{
+	const struct tisci_config *config = dev->config;
+	struct tisci_data *data = dev->data;
+	int ret;
+
+	rx_message.buf = rx_message_buf;
+	rx_message.size = sizeof(rx_message_buf);
+
+	k_sem_init(&response_ready_sem, 0, 1);
+	data->host_id = config->host_id;
+	data->seq = 0x0;
+
+	k_sem_init(&data->data_sem, 1, 1);
+
+	data->desc.max_msg_size = MAILBOX_MBOX_SIZE;
+	/* Conservative timeout */
+	data->desc.max_rx_timeout_ms = 10000;
+
+	ret = mbox_register_callback_dt(&config->mbox_rx, callback, &rx_message);
+
+	if (ret < 0) {
+		LOG_ERR("Could not register callback (%d)\n", ret);
+		return ret;
+	}
+
+	ret = mbox_set_enabled_dt(&config->mbox_rx, true);
+	if (ret < 0) {
+		LOG_ERR("Could not enable RX channel (%d)\n", ret);
+		return ret;
+	}
+	return 0;
+}
+
+/* Clock Management Functions */
+int tisci_cmd_get_clock_state(const struct device *dev, uint32_t dev_id, uint8_t clk_id,
+			      uint8_t *programmed_state, uint8_t *current_state)
+{
+	struct tisci_msg_resp_get_clock_state resp;
+	struct tisci_msg_req_get_clock_state req;
+	struct tisci_xfer *xfer;
+	int ret;
+
+	if (!programmed_state && !current_state) {
+		return -EINVAL;
+	}
+
+	xfer = tisci_setup_one_xfer(dev, TISCI_MSG_GET_CLOCK_STATE, 0, &req, sizeof(req), &resp,
+				    sizeof(resp));
+	if (!xfer) {
+		LOG_ERR("%s:Failed to setup transfer", __func__);
+		return -EINVAL;
+	}
+
+	req.dev_id = dev_id;
+	req.clk_id = clk_id;
+	ret = tisci_do_xfer(dev, xfer);
+
```

Comment:
No newline between a function and the check for its return value. Remove that newline in the couple other spots below also.